### ISSUE TITLE
Unacceptable type of value for attribute "assetDescription"

### ISSUE THREAD

**Issue by Conrad-m-wise** [contentful/contentful.objc#60]
I am getting this crash on trying to sync content (initial).

 *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Unacceptable type of value for attribute: property = "assetDescription"; desired type = NSString; given type = NSNull; value = <null>.'

For an asset with fields:

asset->_localizedFields:
{
    "en-US" =     {
        description = "<null>";
        file =         {
            contentType = "image/jpeg";
            details =             {
                image =                 {
                    height = 400;
                    width = 700;
                };
                size = 102178;
            };
            fileName = "acetategold.jpg";
            url = "//images.contentful.com/0lti2brx2orj/53DH5AHnP24I8UqiOoyICY/45705a5f246ac9b73ccd580df307a8fd/acetategold.jpg";
        };
        title = "Gold Dust Glasses";
    };
}

When i add a description to the asset it no longer fails. This is undesired as the description for an asset is not mandatory.

**Comment by Conrad-m-wise** [comment]
Confirmed this error happens on version 1.10.3 rolling back to version 1.10.2 i do not get this error anymore.

**Comment by neonichu** [comment]
I wasn't really able to reproduce your issue, but I think this change should still resolve it: <https://github.com/contentful/contentful.objc/tree/fix-sanitize-persisted-asset-input> — could you give it a quick spin?

Using

```ruby
pod 'ContentfulDeliveryAPI', :git => 'https://github.com/contentful/contentful.objc', :branch => 'fix-sanitize-persisted-asset-input'
```

in your Podfile should allow testing it.

**Comment by Conrad-m-wise** [comment]
Boris,

I just tested it out and i am not getting the same error. It seems to have fixed the issue.

**Comment by neonichu** [comment]
Thanks for testing, I have merged it and will make a new release soon.

**Comment by neonichu** [comment]
Released as 1.10.4
